dask/delayed.py

```diff
@@ -532,26 +529,18 @@ def __dir__(self):
 
     def __getattr__(self, attr):
         if attr.startswith("_"):
-            raise AttributeError("Attribute {0} not found".format(attr))
+            raise AttributeError(f"Attribute {attr} not found")
 
         if attr == "visualise":
             # added to warn users in case of spelling error
             # for more details: https://github.com/dask/dask/issues/5721
             warnings.warn(
-                "dask.delayed objects have no `visualise` method. Perhaps you meant `visualize`?"
+                "dask.delayed objects have no `visualise` method. "
+                "Perhaps you meant `visualize`?"
             )
 
         return DelayedAttr(self, attr)
 
-    def __setattr__(self, attr, val):
-        if attr in self.__slots__:
-            object.__setattr__(self, attr, val)
-        else:
-            raise TypeError("Delayed objects are immutable")
-
-    def __setitem__(self, index, val):
-        raise TypeError("Delayed objects are immutable")
```

Comment:
Since `DelayedLeaf` and `DelayedAttr` are related to our internal implementation, users might better understand

```python
TypeError: Delayed objects are immutable
```

than

```python
TypeError: 'DelayedLeaf' object does not support item assignment
```

Reply:
fair enough, I'll reinstate it